src/components/Modals/DeleteDocumentModal.jsx

```diff
@@ -0,0 +1,116 @@
+// React Imports
+import React, { useContext } from 'react';
+// Custom Hook Imports
+import { useStatusNotification } from '@hooks';
+// Material UI Imports
+import Button from '@mui/material/Button';
+import ClearIcon from '@mui/icons-material/Clear';
+import CheckIcon from '@mui/icons-material/Check';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+// Utility Imports
+import { runNotification } from '@utils';
+// Context Imports
+import { DocumentListContext } from '@contexts';
+// Component Imports
+import { FormSection } from '../Form';
+
+/**
+ * @typedef {import("../../typedefs.js").deleteDocumentModalProps} deleteDocumentModalProps
+ */
+
+/**
+ * DeleteDocumentModal Component - Component that allows users to delete their
+ * document from the document list stored on their own Pod
+ *
+ * @memberof Modals
+ * @name DeleteDocumentModal
+ * @param {deleteDocumentModalProps} Props - Props for DeleteDocumentModal component
+ * @returns {React.JSX.Element} The DeleteDocumentModal Component
+ */
+const DeleteDocumentModal = ({
+  showDeleteDocumentModal = false,
+  setShowDeleteDocumentModal,
+  selectedDocumentToDelete
+}) => {
+  const { state, dispatch } = useStatusNotification();
+  const { removeDocument } = useContext(DocumentListContext);
+
+  // Event handler for deleting client from client list
+  const handleDeleteDocument = async (event) => {
+    event.preventDefault();
+    runNotification(
+      `Deleting "${selectedDocumentToDelete?.name}" from document list...`,
+      5,
+      state,
+      dispatch
+    );
+    try {
+      await removeDocument(selectedDocumentToDelete?.name);
+    } catch (e) {
+      runNotification(`Document deletion failed. Reason: ${e.message}`);
+    } finally {
+      runNotification(
+        `"${selectedDocumentToDelete?.name}" deleted from document list...`,
+        5,
+        state,
+        dispatch
+      );
+      setTimeout(() => {
+        setShowDeleteDocumentModal(false);
+      }, 2000);
+    }
+  };
+
+  return (
+    <Dialog
+      open={showDeleteDocumentModal}
+      aria-labelledby="dialog-title"
```

Comment:
Maybe not necessarily now but I've been thinking we might want to go through all the different aria labels and make sure they're descriptive and up-to-date. It would help with accessibility, I believe.